addons/sale/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Payam Yasaie <https://www.tashilgostar.com>
+# Copyright 2020 ForgeFlow <http://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def update_account_move_team_partner_ids(cr):
+    openupgrade.logged_query(
+        cr, """
+            UPDATE account_move am
+            SET partner_shipping_id = ai.partner_shipping_id, team_id = ai.team_id
+            FROM account_invoice ai
+            WHERE am.old_invoice_id = ai.id
+            """
+    )
+
+
+def update_sale_order_line_invoice_rel(cr):
+    openupgrade.logged_query(
+        cr, """
+            UPDATE sale_order_line_invoice_rel solir
+            SET invoice_line_id = aml.id
+            FROM account_invoice_line ail
+            JOIN account_move_line aml ON aml.old_invoice_line_id = ail.id
+            WHERE solir.invoice_line_id = ail.id
+            """
+    )
+
+
+def update_product_attribute_custom_value(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE product_attribute_custom_value pacv
+        SET custom_product_template_attribute_value_id = ptav.id
+        FROM product_attribute_value pav
+        JOIN product_template_attribute_value ptav
+            ON ptav.product_attribute_value_id = pav.id
+        WHERE pacv.{} = pav.id
+            AND pacv.custom_product_template_attribute_value_id IS NULL
```

Comment:
You can set product_template_attribute_value with product_template_id != sale_order_id.product_id.product_tmpl_id,
so need rewrite like this:
<pre>
        UPDATE product_attribute_custom_value pacv
        SET custom_product_template_attribute_value_id = ptav.id
        FROM product_attribute_value pav
        JOIN sale_order_line sol ON sol.id = pav.sale_order_line_id
            JOIN product_product pp ON pp.id = sol.product_id
        JOIN product_template_attribute_value ptav
            ON ptav.product_attribute_value_id = pav.id AND ptav.product_tmpl_id = pp.product_tmpl_id
        WHERE pacv.{} = pav.id
            AND pacv.custom_product_template_attribute_value_id IS NULL
</pre>

Reply:
sale.order.line doesn't have a product_template_id column, do it?

Reply:
:facepalm: it's a related non stored field. Changed.